server/src/main/java/jake2/server/SV_MAIN.java

```diff
@@ -711,90 +711,86 @@ static void SV_ExecuteClientMessage(client_t cl) {
         int stringCmdCount = 0;
 
         while (true) {
-            if (Globals.net_message.readcount > Globals.net_message.cursize) {
+            sizebuf_t buffer = Globals.net_message;
+
+            if (buffer.readcount > buffer.cursize) {
                 Com.Printf("SV_ReadClientMessage: bad read:\n");
-                Com.Printf(Lib.hexDump(Globals.net_message.data, 32, false));
+                Com.Printf(Lib.hexDump(buffer.data, 32, false));
                 SV_DropClient(cl);
                 return;
             }
 
-            ClientMessageType c = ClientMessageType.fromInt(MSG.ReadByte(Globals.net_message));
-            ClientMessage msg = ClientMessage.parseFromBuffer(c, Globals.net_message, cl.netchan.incoming_sequence);
-            if (c == ClientMessageType.CLC_BAD) {
-                // todo: warn? drop client?
+            ClientMessage msg = ClientMessage.parseFromBuffer(buffer, cl.netchan.incoming_sequence);
+            if (msg instanceof EndMessage) {
```

Comment:
I thought `CLC_BAD` was something bad, but actually it's just an end of the packet